engine/src/execution_graph/logic_controllers/PhysicalPlanGenerator.h

```diff
@@ -285,11 +285,32 @@ struct tree_processor {
 			cache_settings default_throttled_cache_machine_config = cache_settings{.type = CacheType::SIMPLE, .num_partitions = 1, .context = context->clone(),
 						.flow_control_bytes_threshold = flow_control_bytes_threshold};
 
+			auto child_kernel_type = child->kernel_unit->get_type_id();
+			auto parent_kernel_type = parent->kernel_unit->get_type_id();
 			if (children.size() > 1) {
 				char index_char = 'a' + index;
 				port_name = std::string("input_");
 				port_name.push_back(index_char);
-				if (parent->kernel_unit->can_you_throttle_my_input()){
+
+				if (parent_kernel_type == kernel_type::PartwiseJoinKernel) {
+
+					std::size_t join_partition_size_thresh = 400000000; // 400 MB
```

Comment:
Can you wrap this kind of funcionatily in a macro. This could very easily be a 1 line macro that is something like
BLAZ_GET_CONFIG( SOME_CONSTANT_NAME).

Reply:
Lets improve how we manage the config parameters in a different PR